src/types.rs

```diff
@@ -1534,17 +1757,100 @@ impl Rule {
             None
         }
     }
-    /// Get all the Variables in the Rule.
+    /// Get all the [`Variable`]s in the `Rule`.
+    ///
+    /// [`Variable`]: struct.Variable.html
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_term, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A(x_) = C(x_)").expect("parse of A(x_) = C(x_)");
+    /// let r2 = parse_rule(&mut sig, "B(y_ z_) = C").expect("parse of B(y_ z_) = C");
+    /// 
+    /// let r_variables: Vec<String> = r.variables().iter().map(|v| v.display(&sig)).collect();
+    /// let r2_variables: Vec<String> = r2.variables().iter().map(|v| v.display(&sig)).collect();
+    ///
+    /// assert_eq!(r_variables, vec!["x_"]);
+    /// assert_eq!(r2_variables, vec!["y_", "z_"]);
```

Comment:
To make each example more compact and atomic, I would rewrite this as:

```rust
let r = parse_rule(&mut sig, "A(x_) = C(x_)").expect("parse of A(x_) = C(x_)");
let r_variables: Vec<String> = r.variables().iter().map(|v| v.display(&sig)).collect();
assert_eq!(r_variables, vec!["x_"]);

let r = parse_rule(&mut sig, "B(y_ z_) = C").expect("parse of B(y_ z_) = C");
let r_variables: Vec<String> = r2.variables().iter().map(|v| v.display(&sig)).collect();
assert_eq!(r_variables, vec!["y_", "z_"]);
```

Do you agree?